app/assets/javascripts/application.js

```diff
@@ -133,15 +133,64 @@ $(document).ready(function() {
   //});
   $(".togglelink").click(function() {
     if (typeof $(this).data("toggle") === "undefined") return;
+    // if the grad & school section is disabled then don't allow toggling
+    if (typeof $(".nzdisabled")[0] !== "undefined") return;
     $(this).parent().children("." + $(this).data("toggle"))
       .toggle()
       .prop('disabled', function(i, v) { return !v; });
     
   });
 
+  function set_country_defaults() {
+    $(".nzonly").addClass("nzdisabled");
+
+    // reset jsdisplay for "Add new school"
+    $(".jsdisplay").show().prop('disabled', false);
+    $(".jsnodisplay").hide().prop('disabled', true);
+
+    // set defaults for non-nz
+    $("select.school_select").val($(".school_select option:first").val()); // select no school
+    $("input:radio")[0].checked = true; // select unspecified grad date
+
+    // reset month and year to current date
+    var d = new Date();
+    $("#user_school_graduation_month").val(d.getMonth() + 1);
+    $("#user_school_graduation_year").val(d.getFullYear().toString());
+
+    // disable all fields
+    $(".nzonly input").prop("disabled", true);
+    $(".nzonly select").prop("disabled", true);
+  }
+
+  function on_country_change() {
+    if ($("#togglenz").find(":selected").text() == "New Zealand") {
+      $(".nzonly").removeClass("nzdisabled");
+      $(".nzonly input").prop("disabled", false);
+      $(".nzonly select").prop("disabled", false);
+
+      // reset jsdisplay for "Add new school"
+      $(".jsdisplay").show().prop('disabled', false);
+      $(".jsnodisplay").hide().prop('disabled', true);
+    } else {
+      set_country_defaults();
+    }
+  }
+
+  // legacy jsdisplay behaviour
   $(".jsdisplay").show().prop('disabled', false);
   $(".jsnodisplay").hide().prop('disabled', true);
 
+  // When selecting a country, disable the school and graduation date feilds
+  // for registration and editing a profile
+  if (typeof $("#togglenz")[0] !== "undefined") {
+    $(window).on('load', function(){
+      if ($("#togglenz").find(":selected").text() != "New Zealand") {
+        set_country_defaults();
+      }
+    });
```

Comment:
Unnecessary, since this is all already wrapped in `$(document).ready()`.
```suggestion
    if ($("#togglenz").find(":selected").text() != "New Zealand") {
      set_country_defaults();
    }
```